datafusion/src/execution/context.rs

```diff
@@ -1220,6 +1233,74 @@ impl FunctionRegistry for ExecutionContextState {
     }
 }
 
+/// Task Context Properties
+pub enum TaskProperties {
```

Comment:
I think this change is important (though we could do it as a follow on PR)

Otherwise I was forced to use the following code in https://github.com/influxdata/influxdb_iox/pull/4023 to get the batch size which was quite messy:

```rust
    async fn execute(
        &self,
        _partition: usize,
        context: Arc<TaskContext>,
    ) -> DataFusionResult<SendableRecordBatchStream> {
        let batch_size = {
            if let TaskProperties::SessionConfig(config) = &context.task_settings {
                config.runtime.batch_size
            } else {
                todo!("Need to always have properties")
            }
        };
```

Reply:
I will cover this in the following PR. For batch_size, it will be moved out from RuntimeEnv to SessionConfig.
And SessionConfig just includes the configuration entires, no RuntimeEnv anymore.
And in the TaskContext, there will be a method to get the current SessionConfig.

````
impl TaskProperties {

/// Return the SessionConfig associated with the Task
    pub fn session_config(&self) -> SessionConfig {
        let task_settings = &self.task_settings;
        match task_settings {
            TaskProperties::KVPairs(props) => {
                let session_config = SessionConfig::new();
                session_config
                    .with_batch_size(props.get(BATCH_SIZE).unwrap().parse().unwrap())
                    .with_target_partitions(
                        props.get(TARGET_PARTITIONS).unwrap().parse().unwrap(),
                    )
                    .with_repartition_joins(
                        props.get(REPARTITION_JOINS).unwrap().parse().unwrap(),
                    )
                    .with_repartition_aggregations(
                        props
                            .get(REPARTITION_AGGREGATIONS)
                            .unwrap()
                            .parse()
                            .unwrap(),
                    )
                    .with_repartition_windows(
                        props.get(REPARTITION_WINDOWS).unwrap().parse().unwrap(),
                    )
                    .with_parquet_pruning(
                        props.get(PARQUET_PRUNING).unwrap().parse().unwrap(),
                    )
            }
            TaskProperties::SessionConfig(session_config) => session_config.clone(),
        }
    }
}

````

In this PR, to retrieve the batch size, currently, we can use below code, but this will be changed soon in the following PR.

````
 let batch_size = context.runtime.batch_size();

````